### ISSUE TITLE
dropdown directive bug

### ISSUE THREAD

**Issue by cyberdemon8** [mgcrea/angular-strap#1941]
i've found a bug on bsDropdown directive on line 2873 of dist/angular-strap.js
https://github.com/mgcrea/angular-strap/blob/master/dist/angular-strap.js#L2873

modified from:
```javascript
options.scope = scope;
```
to
```javascript
if (!options) {
  options = {};
}
options.scope = scope;
```
without this correction in certain cases (dropdown in a modal) angular rises this error:
```
TypeError: Cannot set property 'scope' of null
    at postLink (http://localhost:3000/bower_components/angular-strap/dist/angular-strap.js:2873:25)
    at invokeLinkFn (http://localhost:3000/bower_components/angular/angular.js:8281:9)
    at nodeLinkFn (http://localhost:3000/bower_components/angular/angular.js:7791:11)
    at compositeLinkFn (http://localhost:3000/bower_components/angular/angular.js:7140:13)
    at compositeLinkFn (http://localhost:3000/bower_components/angular/angular.js:7143:13)
    at compositeLinkFn (http://localhost:3000/bower_components/angular/angular.js:7143:13)
    at compositeLinkFn (http://localhost:3000/bower_components/angular/angular.js:7143:13)
    at compositeLinkFn (http://localhost:3000/bower_components/angular/angular.js:7143:13)
    at nodeLinkFn (http://localhost:3000/bower_components/angular/angular.js:7786:24)
    at compositeLinkFn (http://localhost:3000/bower_components/angular/angular.js:7140:13)
```

**Comment by edoarsla** [comment]
Having the same issue with this one.  I have the top-of-tree angular-strap.

Plunker:
http://plnkr.co/edit/DvonnSFhxHuKbXupvGbS?p=info

**Comment by cyberdemon8** [comment]
i found a better fix for this directive:
```javascript
directive('bsDropdown', [ '$window', '$sce', '$dropdown', function($window, $sce, $dropdown) {
    return {
      restrict: 'EAC',
      scope: true,
      compile: function(tElement, tAttrs) {
        //removed declaration of var options
        if (!tAttrs.bsDropdown) {
          var nextSibling = tElement[0].nextSibling;
          while (nextSibling && nextSibling.nodeType !== 1) {
            nextSibling = nextSibling.nextSibling;
          }
          if (nextSibling.classList.contains('dropdown-menu')) {
            tAttrs.template = nextSibling.outerHTML; //replaced options with tAttrs
            tAttrs.templateUrl = undefined; //replaced options with tAttrs
            nextSibling.parentNode.removeChild(nextSibling);
          }
        }
        return function postLink(scope, element, attr) {
          //declare var options and set scope
          var options = {
            scope: scope
          };
          angular.forEach([ 'template', 'templateUrl', 'controller', 'controllerAs', 'placement', 'container', 'delay', 'trigger', 'keyboard', 'html', 'animation', 'id' ], function(key) {
            if (angular.isDefined(tAttrs[key])) options[key] = tAttrs[key];
          });
          var falseValueRegExp = /^(false|0|)$/i;
          angular.forEach([ 'html', 'container' ], function(key) {
            if (angular.isDefined(attr[key]) && falseValueRegExp.test(attr[key])) options[key] = false;
          });
          attr.bsDropdown && scope.$watch(attr.bsDropdown, function(newValue, oldValue) {
            scope.content = newValue;
          }, true);
          attr.bsShow && scope.$watch(attr.bsShow, function(newValue, oldValue) {
            if (!dropdown || !angular.isDefined(newValue)) return;
            if (angular.isString(newValue)) newValue = !!newValue.match(/true|,?(dropdown),?/i);
            newValue === true ? dropdown.show() : dropdown.hide();
          });
          var dropdown = $dropdown(element, options);
          scope.$on('$destroy', function() {
            if (dropdown) dropdown.destroy();
            options = null;
            dropdown = null;
          });
        };
      }
    };
  }
```

**Comment by edoarsla** [comment]
Thank you Valentino, this fixes the problem I was experiencing.

**Comment by mgcrea** [comment]
Fixed by #1947
